src/jdk.jpackage/share/classes/jdk/jpackage/internal/Arguments.java

```diff
@@ -478,9 +474,27 @@ private static void nextArg() {
             context().pos++;
         }
 
+        private static void prevArg() {
+            Objects.checkIndex(context().pos, context().argList.size());
+            context().pos--;
+        }
+
         private static boolean hasNextArg() {
             return context().pos < context().argList.size();
         }
+
+        private static Runnable createArgumentWithOptionalValueAction(String option) {
```

Comment:
I am not sure, but did you test case when `--win-menu` or `--win-shortcut` or `--linux-shortcut` is last argument without value? With new code `var value = popArg();` will be `""` and we will call `setOptionValue(option, value)`. Before change it will be set to `true`.

Reply:
You are right. I reworked the tests, and they start failing if the last argument is `--win-menu`, `--win-shortcut`, or `--linux-shortcut`.